### PR TITLE
Docs/docker instructions

### PR DESCRIPTION
**Pull request by garyluu**
Started editing the docker page then became random edits here and there

### CODE REVIEW

File: faq.md [hunk 1]
```diff
@@ -86,14 +86,14 @@ Additionally:
...
+Docker behaves a bit differently on a [Mac](https://docs.docker.com/docker-for-mac/osxfs/#/namespaces) than on a typical ubuntu machine. By default the only shared volumes are /Users, /Volumes, /tmp, and /private. Note that /var is not a shared directory (and can't be set as one). `cwltool` uses your TMPDIR (the env variable) to setup volumes with docker, which on a Mac can default to a subdirectory of /var. In order to get `cwltool` working on your Mac, you need to set your TMPDIR to be under one of the shared volumes in Docker for Mac. You can do this by doing something similar to the following:
```
Comment (coverbeck) [inline faq.md]: **U**buntu

File: _docs/advanced-features.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ In the following JSON file, this file indicates for a CWL run that the input fil
 }
 ```
 
-The Dockstore command-line allows you to specify that the input file can be at a HTTP(S) location, an FTP location, an AWS S3 location, a [synapse id](http://python-docs.synapse.org/#accessing-data), or an [ICGC storage id](http://docs.icgc.org/cloud/guide/#cloud-guide) in place of that path. For example the following indicates that the input file will be downloaded under HTTP.
+The Dockstore command-line allows you to specify that the input file can be at an HTTP(S) location, an FTP location, an AWS S3 location, a [synapse id](http://python-docs.synapse.org/#accessing-data), or an [ICGC storage id](http://docs.icgc.org/cloud/guide/#cloud-guide) in place of that path. For example, the following indicates that the input file will be downloaded under HTTP.
```
[the remark this answers is not in the document]
Comment (denis-yuen) [reply _docs/advanced-features.md]: I think its worth noting both DOS and the fact that its more of a pointer.

File: _docs/checker-workflows.md [hunk 3]
```diff
@@ -81,11 +81,11 @@ The first step is to find the tool under the my tools page. In the info tab ther
 When registering a checker workflow, you need the following fields:
 * Default checker workflow path (path to main descriptor of the checker workflow)
 * Default test parameter file (if not given will copy over from original entry)
-* Descriptor type (cwl or wdl)
+* Descriptor type (CWL or WDL) when original entry is a tool
```
[the remark this answers is not in the document]
Comment (denis-yuen) [reply _docs/checker-workflows.md]: Nope, not yet anyway

File: _docs/advanced-features.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (garyluu) [reply _docs/advanced-features.md]: DOS was added in after merge conflict resolution

**Pull request merged**